pkg/resources/pkey_selector.go

```diff
@@ -0,0 +1,27 @@
+package resources
+
+import (
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/types"
+)
+
+type pkeySelector struct {
+	pKeys []string
+}
+
+// NewPKeySelector returns a DeviceSelector interface to filter devices based on available PKeys
+func NewPKeySelector(pKeys []string) types.DeviceSelector {
+	return &pkeySelector{pKeys: pKeys}
+}
+
+func (ds *pkeySelector) Filter(inDevices []types.HostDevice) []types.HostDevice {
+	filteredList := make([]types.HostDevice, 0)
+
+	for _, dev := range inDevices {
+		pKey := dev.(types.PciNetDevice).GetPKey()
+		if pKey != "" && contains(ds.pKeys, pKey) {
```

Comment:
could we keep the names consistent? 
pKeys in some places, pkey in others, and PKeys in another.

Reply:
Thanks for noticing! I've changed lowercase pkey to the snake case. pKey/PKey are both snake case and determined by the context